uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Service/CiviHRStatsGatherer.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+use CRM_HRCore_Model_CiviHRStatistics as CiviHRStatistics;
+use CRM_HRCore_Model_ReportConfiguration as ReportConfiguration;
+use CRM_HRCore_Model_ReportConfigurationAgeGroup as AgeGroup;
+use CRM_HRCore_CMSData_Variable_VariableServiceInterface as VariableServiceInterface;
+use CRM_HRCore_Helper_ExtensionHelper as ExtensionHelper;
+use CRM_HRCore_CMSData_Variable_DrupalVariableService as DrupalVariableService;
+
+/**
+ * Responsible for gathering all required site statistics that will be sent to
+ * monitor site usage.
+ */
+class CRM_HRCore_Service_CiviHRStatsGatherer {
+
+  /**
+   * @var VariableServiceInterface
+   */
+  protected $cmsVariableService;
+
+  /**
+   * @param VariableServiceInterface $cmsVariableService
+   */
+  public function __construct(VariableServiceInterface $cmsVariableService) {
+    $this->cmsVariableService = $cmsVariableService;
+  }
+
+  /**
+   * Fetch and set all required statistics.
+   *
+   * @return CiviHRStatistics
+   */
+  public function gather() {
+    $stats = new CiviHRStatistics();
+    $stats->setGenerationDate(new \DateTime());
+    $stats->setSiteName($this->cmsVariableService->get('site_name'));
+    $this->setBaseUrl($stats);
+    $this->setEntityCounts($stats);
+    $this->setContactSubtypes($stats);
+    $this->setReportConfigurations($stats);
+    $this->setAgeGroups($stats);
+
```

Comment:
I can see that this method is not gathering all the data described on the wiki. Mainly, it's missing information of the number of users per role and the login time information. Shouldn't we do that here?

Reply:
Oops. This PR is big enough already so I created PCHR-2599 for this.

Reply:
Big enough how? I don't mind having a separate ticket for that, but I don't understand the reason. Why don't we have then one separate ticket for each different type of statistics?

Reply:
We're gone over the expected time limit. I agree it could have been broken down better, which is what I'm suggesting now.